src/hooks/useCalculateSwapPairs.ts

```diff
@@ -99,11 +99,11 @@ export function useCalculateSwapPairs(): (token?: Token) => SwapData[] {
   }, [chainId])
 
   return useCallback(
-    function calculateSwapPairs(token?: Token): SwapData[] {
+    function calculateSwapPairs(token?: BasicToken): SwapData[] {
       if (token == null || chainId == null || tokens === null) return []
       const cacheHit = pairCache[token.symbol]
       if (cacheHit) return cacheHit
-      const originTokenAddress = (token.addresses[chainId] || "").toLowerCase()
+      const originTokenAddress = (token?.address || "").toLowerCase()
```

Comment:
The token nullcheck already happened above. Don't think you need the optional here.